packages/dicom/gdcm/DICOMTagReader.h

```diff
@@ -0,0 +1,582 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef DICOM_TAG_READER_H
+#define DICOM_TAG_READER_H
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <string.h>
+
+#include <iconv.h>
+
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+
+#include "itkCommonEnums.h"
+#include "itkGDCMImageIO.h"
+#include "itkGDCMSeriesFileNames.h"
+#include "itkImageIOBase.h"
+#include "itkMetaDataObject.h"
+
+const std::string DEFAULT_ENCODING("ISO_IR 6");
+const std::string DEFAULT_ISO_2022_ENCODING("ISO 2022 IR 6");
+constexpr const char *ASCII = "ASCII";
+
+// delimiters: CR, LF, FF, ESC, TAB (see
+//  https://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_6.1.3,
+//  table 6.1-1)
+// Also includes 05/12 (BACKSLASH in IR 13 or YEN SIGN in IR 14), since that
+//  separates Data Element Values and it resets to initial charset.
+//  See: dicom part 5, sect 6.1.2.5.3
+constexpr const char *DEFAULT_DELIMS = "\x1b\x09\x0a\x0c\x0d\x5c";
+// DEFAULT_DELIMS + "^" and "="
+constexpr const char *PATIENT_NAME_DELIMS = "\x1b\x09\x0a\x0c\x0d\x5c^=";
+
+std::string
+unpackMetaAsString(const itk::MetaDataObjectBase::Pointer &metaValue)
+{
+    using MetaDataStringType = itk::MetaDataObject<std::string>;
```

Comment:
Switch by tag and expose as a more accurate type? https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/GDCM/src/itkGDCMImageIO.cxx#L977